src/tools/moveedge.cpp

```diff
@@ -0,0 +1,32 @@
+#include "moveedge.h"
+#include "editorview.h"
+#include "edgeitem.h"
+#include <QMouseEvent>
+
+MoveEdgeTool::MoveEdgeTool()
+: BaseTool()
+{
+}
+
+bool MoveEdgeTool::mousePressEvent(EditorView* editor, QMouseEvent* event)
+{
+  for (EdgeItem* item : editor->selectedItems<EdgeItem>()) {
+    item->setSelected(false);
+  }
+  QList<EdgeItem*> gripsInRing = editor->itemsInRing<EdgeItem>();
+  for (EdgeItem* item : gripsInRing)
+  {
+    item->setSelected(true);
+  }
+  return true;
+}
+
+bool MoveEdgeTool::mouseMoveEvent(EditorView* editor, QMouseEvent* event)
+{
+  return true;
```

Comment:
This works...? I would have expected you to need to return false.

Reply:
It doesn't cause we don't have a way to move edges yet. I return true so it would drag vertices for now. This is very much a todo class.